### PR TITLE
changes to high resolution timer, fixes #16

### LINKED ISSUE #16: high resolution timer cannot be changed realtime
With the winmm implementation, the high resolution timer could be changed on the fly. During the rewrite to ntdll, this is no longer possible. 

https://github.com/ReactiveDrop/reactivedrop_public_src/blob/reactivedrop_public/src/game/server/highres_timer.cpp

Expected result:

changing rd_dedicated_high_resolution_timer to be applied realtime when changed, or when the first player joins.

Actual result:

changing rd_dedicated_high_resolution_timer is locked after first player joins and cannot be altered after.

### PR DESCRIPTION
**Pull request by mithrand0**
…op/reactivedrop_public_src/issues/16

### CODE REVIEW

File: src/game/shared/swarm/asw_gamerules.cpp [hunk 1]
```diff
@@ -180,10 +180,18 @@ extern ConVar old_radius_damage;
 	ConVar rd_sentry_block_aliens( "rd_sentry_block_aliens", "1", FCVAR_CHEAT, "If 0 sentries don't collide with aliens" );
 	ConVar rd_auto_fast_restart( "rd_auto_fast_restart", "0", FCVAR_NONE, "Set to 1 to restart mission on fail automatically" );
 	ConVar rd_adjust_mod_dont_load_vertices("rd_adjust_mod_dont_load_vertices", "1", FCVAR_NONE, "Automatically disables loading of vertex data.", true, 0, true, 1);
-	ConVar rd_high_resolution_timer_ms ( "rd_dedicated_high_resolution_timer_ms", "0.1", FCVAR_NONE, "Acquire a high resolution timer with specified resolution." );
+	ConVar rd_high_resolution_timer_ms( "rd_dedicated_high_resolution_timer_ms", "0.01", FCVAR_NONE, "Acquire timer with specified resolution in ms" );
```
Comment (geotavros) [inline src/game/shared/swarm/asw_gamerules.cpp:183]: Is there a reason why ConVar in code rd_high_resolution_timer_ms differs in name with the console version rd_dedicated_high_resolution_timer_ms?

Comment (mithrand0) [reply src/game/shared/swarm/asw_gamerules.cpp]: No reason. Just pushed a commit to sync the names.

File: src/game/server/highres_timer.cpp [hunk 2]
```diff
@@ -4,16 +4,32 @@
...
-	{
-		ULONG requestRes = ms * 1000;
 		ULONG currentRes;
+		ULONG desiredRes = 15.0f; // 15 ms os default
+		bool apply = true;
 
-		newRes = NtSetTimerResolution( round(ms * 1000), TRUE, &currentRes );
-		DevMsg( "timer resolution changed from %d to %d ns\n", currentRes, newRes );
+		// if zero or negative, restore default os timer resolution
+		if ( ms > 0.00001f ) {
```
Comment (BenLubar) [inline src/game/server/highres_timer.cpp:20]: should this be `>=`?

**Comment by BenLubar** [comment]
What happens if someone sets this very high, for example to 1000?

**Pull request merged**